v2/web/src/pages/home/components/birthday-dialog.vue

```diff
@@ -0,0 +1,190 @@
+                <img :src="randomMessage('birthdayMatchImages')" alt="confetti" />
+            </template>
+
+            <template v-else>
+                <h2>{{ randomMessage('birthdayNoMatch') }}</h2>
+                <img :src="randomMessage('birthdayNoMatchImages')" alt="confetti" />
+            </template>
+        </div>
+
+        <div class="dialog-body">
+            <birthday-input
+                v-if="!birthdayIsSet"
+                @change:day="setDay"
+                @change:month="setMonth"
+            />
+        </div>
+
+        <div class="dialog-footer">
+            <button
+                v-if="!birthdayIsSet"
+                class="btn btn-primary"
+                ref="submitButton"
+                @click="submitBirthday"
+            >
+                {{ todaysMessage }}
+            </button>
+
+            <router-link
+                v-else
+                to="/signup"
+                class="sign-up-message"
+            >
+                Sign up to get on the leaderboard!
+            </router-link>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import { getBirthday } from '@/utils/getBirthday';
+import { random } from '@/utils/seededRandom';
+import PopupMessages from '@/copy/popup-messages.json';
+import BirthdayInput, { getMonthNumber } from '@/shared/birthday-input.vue';
+import type { Months } from '@/shared/birthday-input.vue';
+
+const userBirthday = await getBirthday();
+
+const emit = defineEmits(['close-dialog']);
+
+const todaysMessage = PopupMessages.birthdayButton[new Date().getDay()];
+
+const dayInput = ref<HTMLInputElement | null>(null);
+const submitButton = ref<HTMLButtonElement | null>(null);
+
+const birthday = ref({ 
+    month: userBirthday?.month || 6,
+    day: userBirthday?.day || 15,
+});
+
+const birthdayIsSet = ref(userBirthday !== null);
+const todayIsBirthday = computed(() => {
+    if (!birthdayIsSet.value) return false;
+
+    const today = new Date();
+    return birthday.value?.month === today.getMonth() + 1 && birthday.value?.day === today.getDate();
+});
+
+function setMonth(nextValue: Months) {
+    const monthCode = getMonthNumber(nextValue);
+
+    if (!birthday.value) {
+        birthday.value = {
+            month: monthCode,
+            day: 0, // 0 is falsy and an invalid day
+        };
+    } else {
+        birthday.value.month = monthCode;
+    }
+
+    dayInput.value?.focus();
+}
+
+function setDay(nextValue: string) {
+    const day = parseInt(nextValue);
+
+    if (!birthday.value) {
+        birthday.value = {
+            month: 0, // 0 is falsy and an invalid month
+            day,
+        };
+    } else {
+        birthday.value.day = day;
+    }
+}
+
+function submitBirthday() {
+    if (birthday.value) {
+        birthdayIsSet.value = true;
+        // localStorage.setItem('birthday', `${birthday.value.month}-${birthday.value.day}`);
```

Comment:
Re-enable this

Reply:
Resolved with [f5e1abb](https://github.com/Ross-Alexandra/is-it-my-birthday-app/pull/21/commits/f5e1abb6518c012f857c8c4df5badcaaab91886a)